server/src/data_store.cpp

```diff
@@ -251,13 +243,13 @@ bool DataStore::ReadDistinctMacAddrs(const std::string& device_id, std::vector<s
         {
             console_->error("Failed to read from database");
             sqlite3_finalize(selectStmt);
-            return false;
+            return {};
```

Comment:
I think these lines have some repeating code and therefore can be simplified.

Specifically you can move the debug print of line 251 over the break at line 240.
Then replace the `sqlite3_finalize` and `return {}` statements at lines 245 and 246 with a simple `break;`.

Finally, on the end of the function you always `return mac_addrs` and you don't need to care whether it is empty or not. :+1:

Reply:
Smart!
"Specifically you can move the debug print of line 251 over the break at line 240."   - Why should the debug print of 251 be moved though?

Reply:
So you don't have both the printouts, i.e. "failed to read bla bla" AND the other one that would "always" be executed in the end before the return.